pkg/storage/tsdb/ref_cache.go

```diff
@@ -0,0 +1,161 @@
+package tsdb
+
+import (
+	"sync"
+	"time"
+
+	"github.com/cortexproject/cortex/pkg/ingester/client"
+	"github.com/cortexproject/cortex/pkg/util"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/pkg/labels"
+)
+
+const (
+	// DefaultRefCacheTTL is the default RefCache purge TTL. We use a reasonable
+	// value that should cover most use cases. The cache would be ineffective if
+	// the scrape interval of a series is greater than this TTL.
+	DefaultRefCacheTTL = int64(5 * time.Minute)
+
+	refCacheStripes = 128
+)
+
+// RefCache is a single-tenant cache mapping a labels set with the reference
+// ID in TSDB, in order to be able to append samples to the TSDB head without having
+// to copy write request series labels each time (because the memory buffers used to
+// unmarshal the write request is reused).
+type RefCache struct {
+	// The cache is splitted into stripes, each one with a dedicated lock, in
+	// order to reduce lock contention.
+	stripes map[uint8]*refCacheStripe
```

Comment:
Since stripes isn't being expanded I think we should use an array (since we have a const) instead of a map.

Reply:
Oh sure! I'm wondering why I've used a map here 🤔